collectors/proc.plugin/sys_class_drm.c

```diff
@@ -0,0 +1,1148 @@
+        rrdset_is_obsolete(c->st_mem_usage_vis_vram);
+        return 1;
+    }
+}
+
+static int do_rrd_gtt(struct card *const c){
+    if(likely(!read_single_number_file(c->pathname_mem_used_gtt, (unsigned long long *) &c->used_gtt) && 
+            c->total_gtt)){
+        rrddim_set_by_pointer(  c->st_mem_usage_perc_gtt, 
+                                c->rd_mem_used_perc_gtt, 
+                                c->used_gtt * 10000 / c->total_gtt);
+        rrdset_done(c->st_mem_usage_perc_gtt);
+    
+        rrddim_set_by_pointer(c->st_mem_usage_gtt, c->rd_mem_used_gtt, c->used_gtt);
+        rrddim_set_by_pointer(c->st_mem_usage_gtt, c->rd_mem_free_gtt, c->total_gtt - c->used_gtt);
+        rrdset_done(c->st_mem_usage_gtt);
+        return 0;
+    }
+    else {
+        collector_error("Cannot read used_gtt for %s: [%s]", c->pathname, c->id.marketing_name);
+        freez((void *) c->pathname_mem_used_gtt);
+        freez((void *) c->pathname_mem_total_gtt);
+        rrdset_is_obsolete(c->st_mem_usage_perc_gtt);
+        rrdset_is_obsolete(c->st_mem_usage_gtt);
+        return 1;
+    }
+}
+
+static char *set_id(const char *suf_1, const char *suf_2, const char *suf_3){
+    static char id[RRD_ID_LENGTH_MAX + 1];
+    snprintfz(id, RRD_ID_LENGTH_MAX, "%s_%s_%s", suf_1, suf_2, suf_3);
+    return id;
+}
+
+int do_sys_class_drm(int update_every, usec_t dt) {
+    (void)dt;
+    
+    static DIR *drm_dir = NULL;
+
+    int chart_prio = NETDATA_CHART_PRIO_DRM_AMDGPU;
+
+    if(unlikely(!drm_dir)) {
+        char filename[FILENAME_MAX + 1];
+        snprintfz(filename, FILENAME_MAX, "%s%s", netdata_configured_host_prefix, "/sys/class/drm");
+        char *drm_dir_name = config_get(CONFIG_SECTION_PLUGIN_PROC_DRM, "directory to monitor", filename);
+        if(unlikely(NULL == (drm_dir = opendir(drm_dir_name)))){
+            collector_error("Cannot read directory '%s'", drm_dir_name);
+            return 1;
+        }
+
+        struct dirent *de = NULL;
+        while(likely(de = readdir(drm_dir))) {
+            if( de->d_type == DT_DIR && ((de->d_name[0] == '.' && de->d_name[1] == '\0') || 
+                 (de->d_name[0] == '.' && de->d_name[1] == '.' && de->d_name[2] == '\0'))) continue;
+            
+            if(de->d_type == DT_LNK && !strncmp(de->d_name, "card", 4) && !strchr(de->d_name, '-')) {
+                struct card *const c = callocz(1, sizeof(struct card));  
+
+                /* Get static info */              
+
+                char filename[FILENAME_MAX + 1];
+                snprintfz(filename, FILENAME_MAX, "%s/%s", drm_dir_name, de->d_name);
+                c->pathname = strdupz(filename);
+
+                snprintfz(filename, FILENAME_MAX, "%s/%s", c->pathname, "device/device");
+                if(read_single_base64_or_hex_number_file(filename, (unsigned long long *) &c->id.asic_id)){
+                    collector_error("Cannot read asic_id from '%s'", filename);
```

Comment:
I think all these errors are not errors. But debug or info at least.

Having AMD GPU is absolutely not required so failing to read those files is not an error but expected.

If we keep them as it is **all** users w/o AMD GPU will get potentially NumOfProps*NumberOfCards (in `/sys/class/drm`) error log lines on every restart.